pkg/pgmodel/lreader/labels_reader.go

```diff
@@ -97,41 +92,39 @@ func (lr *labelsReader) LabelNames() ([]string, error) {
 	return labelNames, nil
 }
 
-// PrompbLabelsForIds returns protobuf representation of the label sets for
-// the provided label ids
-func (lr *labelsReader) PrompbLabelsForIds(ids []int64) (lls []prompb.Label, err error) {
-	ll, err := lr.LabelsForIds(ids)
-	if err != nil {
-		return
-	}
-	lls = make([]prompb.Label, len(ll))
-	for i := range ll {
-		lls[i] = prompb.Label{Name: ll[i].Name, Value: ll[i].Value}
+func (lr *labelsReader) LabelsForIdMap(idMap map[int64]*labels.Label) (err error) {
+	numIds := len(idMap)
+	keys := make([]interface{}, numIds)
+	values := make([]interface{}, numIds)
+
+	_, present := idMap[0]
+	if present {
+		return fmt.Errorf("Looking up a label for id 0")
```

Comment:
Why do we error for `0` label id? I remember in some tests that I had 0 as id which indicated they were not created if I recall. But should we be erroring for this?

Reply:
If it's required, then a comment will be helpful.

Reply:
added a comment